CONTRIBUTING.md

```diff
@@ -0,0 +1,78 @@
+
+## Build instructions
+
+### Setup
+
+This project is built using [poetry](https://python-poetry.org/). To learn more about the basics of working with this tool, read [Poetry's basic usage guide](https://python-poetry.org/docs/basic-usage/).
+
+To begin development, active the poetry shell and ensure your dependencies are installed.
+
+```
+poetry shell
+poetry install
+```
+
+### Testing
+
+To run all unit tests:
+
+```shell
+make test
+```
+
+It is preferable to run tests against all supported minor versions of Python (as described in `README.md` under Requirements), or at least the lowest and highest versions, prior to submitting a pull request. However, LaunchDarkly's CI tests will run automatically against all supported versions.
+
+### Building documentation
+
+See "Documenting types and methods" below. To build the documentation locally, so you can see the effects of any changes before a release:
+
+```shell
+make docs
+```
+
+The output will appear in `docs/build/html`. Its formatting will be somewhat different since it does not have the same stylesheets used on readthedocs.io.
+
+### Running the linter
+
+The `mypy` tool is used in CI to verify type hints and warn of potential code problems. To run it locally:
+
+```shell
+make lint
+```
+
+## Code organization
+
+The library's module structure is as follows:
+
+* `ldotel.tracing`: This module contains entry points for OTEL features related to tracing.
+
+A special case is the module `ldotel.impl`, and any modules within it. Everything under `impl` is considered a private implementation detail: all files there are excluded from the generated documentation, and are considered subject to change at any time and not supported for direct use by application developers. Alternately, class names can be prefixed with an underscore to be "private by convention"; that will at least prevent them from being included in wildcard imports like `from ldotel import *`, but it is still preferable to avoid a proliferation of implementation-only modules within the main `ldotel` module, since developers may wrongly decide to reference such modules in imports.
+
+So, if there is a class whose existence is entirely an implementation detail, it should be in `impl`. Similarly, classes that are _not_ in `impl` must not expose any public members (i.e. symbols that do not have an underscore prefix) that are not meant to be part of the supported public API. This is important because of our guarantee of backward compatibility for all public APIs within a major version: we want to be able to change our implementation details to suit the needs of the code, without worrying about breaking a customer's code. Due to how the language works, we can't actually prevent an application developer from referencing those classes in their code, but this convention makes it clear that such use is discouraged and unsupported.
```

Comment:
```suggestion
So, if there is a class whose existence is entirely an implementation detail, it should be in `impl`. Similarly, classes that are  in `impl` must only expose public members (i.e. symbols that do not have an underscore prefix) that are meant to be part of the supported public API. This is important because of our guarantee of backward compatibility for all public APIs within a major version: we want to be able to change our implementation details to suit the needs of the code, without worrying about breaking a customer's code. Due to how the language works, we can't actually prevent an application developer from referencing those classes in their code, but this convention makes it clear that such use is discouraged and unsupported.
```

I tried to remove the double negative, but might have messed up the meaning. I think it is right.